### PR TITLE
ovl_Npc_Engineer OK

### PR DESCRIPTION
**Pull request by MegaIDK**

### CODE REVIEW

File: src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c [hunk 1]
```diff
@@ -1,11 +1,12 @@
 #include "ac_npc_engineer.h"
 #include "m_lib.h"
 #include "overlays/gamestates/ovl_play/m_play.h"
+#include "include/unknown_structs.h"
 
 void aNEG_actor_ct(Actor* thisx, Game_Play* game_play);
-void aNEG_actor_dt(Actor* thisx, Game_Play* game_play);
-void aNEG_actor_init(Actor* thisx, Game_Play* game_play);
-void aNEG_actor_save(Actor* thisx, Game_Play* game_play);
+void aNEG_actor_dt();
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c:7]: These are still `ActorFunc` typed so they still take those args even if unused

File: src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c [hunk 2]
```diff
@@ -23,14 +24,43 @@ ActorProfile Npc_Engineer_Profile = {
...
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer/func_809AEF54_jp.s")
+void aNEG_actor_ct(Actor* thisx, Game_Play* game_play) {
+    Npc_Engineer* this = (Npc_Engineer*)thisx;
+    if (common_data.unk_1004C->unk_BC(thisx, game_play) == 1) {
+        this->unk7C0 = none_proc1;
+        common_data.unk_1004C->unk_C0(this, game_play, &D_809AEFA4_jp);
+        this->unk80C = -1;
+        this->unk7FD = 0;
+        this->unk8AC = -1;
+        this->actor.colStatus.mass = 0xFE;
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c:38]:
```suggestion
        this->actor.colStatus.mass = MASS_HEAVY ;
```

File: src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.h [hunk 3]
```diff
@@ -12,7 +12,21 @@ typedef void (*Npc_EngineerActionFunc)(struct Npc_Engineer*, struct Game_Play*);
...
+    /* 0x7D6 */ u8 unk7D6;
+    /* 0x174 */ UNK_TYPE1 unk_7D7[0x26];
+    /* 0x7FD */ u8 unk7FD;
+    /* 0x174 */ UNK_TYPE1 unk_7FE[0xE];
+    /* 0x80C */ u32 unk80C;
+    /* 0x174 */ UNK_TYPE1 unk_810[0x9C];
+    /* 0x8AC */ s32 unk8AC;
+    /* 0x174 */ UNK_TYPE1 unk_8B0[0x2C];
+    /* 0x8DC */ s16 unk8DC;
+    /* 0x8DE */ UNK_TYPE1 unk_8DE[0x5A];
```
Comment (Prakxo) [inline src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.h:29]: Fix the offsets

File: src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c (hunk 1 above)
Comment (MegaIDK) [reply src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c:7]: it doesnt match with them there though or at least not with them when they get defined underneath

Comment (AngheloAlf) [reply src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c:9]: Try passing the missing args to the called function, that may help.
If you have problems don't doubt in asking for help

Comment (MegaIDK) [reply src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c:9]: yeah it matched when i did that lmao 
i mean i didnt ask about it cus ig i stumbled upon like a wrong way of doing it that looked right so there didnt seem to have a problem, ill def ask before idk doing a dumb like this again lol

File: src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c [hunk 4]
```diff
@@ -25,14 +26,43 @@ ActorProfile Npc_Engineer_Profile = {
 };
 #endif
 
-#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer/aNEG_actor_ct.s")
+//#pragma GLOBAL_ASM("asm/jp/nonmatchings/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer/aNEG_actor_ct.s")
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c:29]: Could you remove the GLOBAL_ASM lines now that the functions are matched please?

File: src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c [hunk 5]
```diff
@@ -25,14 +26,43 @@ ActorProfile Npc_Engineer_Profile = {
...
+extern struct_809AEFA4 D_809AEFA4_jp;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c:30]: Don't forget to migrate the data to the C file. Currently the data is located at `asm/jp/data/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.data.s`. If you have problems migrating the data feel free to ask

File: include/m_common_data.h [hunk 6]
```diff
@@ -29,6 +29,15 @@ typedef UNK_PTR (*CommonData_unk_1004C_unk_0C)(size_t, const struct struct_80116
 typedef UNK_RET (*CommonData_unk_1004C_unk_10)(struct Actor*);
 typedef UNK_RET (*CommonData_unk_1004C_unk_14)(struct CommonData_unk_1004C_unk_14_arg0*, u16);
 
+typedef UNK_RET (*CommonData_unk_1004C_unk_BC)(struct Actor*, struct Game_Play*);
+typedef void (*CommonData_unk_1004C_unk_C0)(struct Actor*, struct Game_Play*, void*);
```
Comment (AngheloAlf) [inline include/m_common_data.h:33]:
```suggestion
typedef void (*CommonData_unk_1004C_unk_C0)(struct Actor*, struct Game_Play*, struct struct_809AEFA4*);
```

You'll also need to put a `struct struct_809AEFA4;` at the top of this file

File: src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c [hunk 7]
```diff
@@ -20,19 +22,51 @@ ActorProfile Npc_Engineer_Profile = {
...
+struct_809AEFA4 D_809AEFA4_jp = {
+    func_809AEF28_jp,
+    func_809AEF54_jp,
+    4,
+    0,
+};
+
+void aNEG_actor_ct(Actor* thisx, Game_Play* game_play) {
+    Npc_Engineer* this = (Npc_Engineer*)thisx;
+    if (common_data.unk_1004C->unk_BC(thisx, game_play) == 1) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Npc_Engineer/ac_npc_engineer.c:38]: Could you leave an empty line between the declaration and the rest of the code?
```suggestion
    Npc_Engineer* this = (Npc_Engineer*)thisx;

    if (common_data.unk_1004C->unk_BC(thisx, game_play) == 1) {
```

**Review by AngheloAlf — APPROVED** [review]

File: include/m_common_data.h (hunk 6 above)
Comment (Kelebek1) [reply include/m_common_data.h]: It may not be shared?

**Pull request merged**